system-tests/test.py

```diff
@@ -227,23 +228,26 @@ class RunGrab(EnterPython):
     """
     Run Fab grab tool against a repository.
     """
-    def __init__(self, test_directory: Path, scheme: str):
-        self._scheme = scheme
+    def __init__(self, test_directory: Path, repo: str, protocol: str):
+        self._scheme = f"{repo}+{protocol}"
         self._repo_path = test_directory.absolute() / "repo"
-        self._svn_server: Optional[subprocess.Popen] = None
+        self._server: Optional[subprocess.Popen] = None
 
-        if scheme == 'file':
-            repo_url = f'file:////{self._repo_path}'
-        elif scheme == 'svn':
-            repo_url = 'svn://127.0.0.1/'
-        elif scheme == 'http':
+        if protocol == 'http':
             # TODO: This scheme is included for completeness. Currently there
-            #       is no obvious way to test this with out an Apache server.
-            #       Which is way too much to consider at the moment.
+            #       is no obvious way to test this without an Apache server
+            #       which is way too much to consider at the moment.
             # repo_url = f'http://localhost/repo'
-            raise Exception("Unable to test Subversion over HTTP protocol.")
+            message = "Unable to test Fetch over HTTP protocol."
+            raise NotImplementedError(message)
+
+        repo_url = f'{self._scheme}://'
+        if protocol == 'file':
+            repo_url += f'//{self._repo_path}'
+        elif protocol in ['git', 'http', 'svn']:
```

Comment:
Perhaps `http` shouldn't be included here if it is excepted just above?

Reply:
While the code is correct, HTTP would require a hostname, I don't care strongly enough to argue. I've removed it.